boards/cubepilot/cubeorangeplus/src/board_config.h

```diff
@@ -44,6 +44,16 @@
 #include <stdint.h>
 #include <stm32_gpio.h>
 
+/**
+ * If NuttX is built without support for SMPS it can brick the hardware.
+ * Therefore, we make sure the NuttX headers are correct.
+ */
+#include "hardware/stm32h7x3xx_pwr.h"
+#if STM32_PWR_CR3_SMPSEXTHP != (1 << 3)
```

Comment:
Is there a reason (historical, or otherwise) you need to check the particular value rather than `CONFIG_STM32H7_HAVE_SMPS` (and avoid including the pwr header)?

Reply:
As I understand it you will have the config define but if you haven't updated the NuttX submodule, then it still won't actually do anything, and brick it. So you really need to check whether the implementation (e.g. the define) is there.